src/fusor/translator.py

```diff
@@ -0,0 +1,580 @@
+"""Module for translating output from fusion detection methods to FUSOR AssayedFusion
+objects
+"""
+
+import logging
+from enum import Enum
+
+import polars as pl
+from cool_seq_tool.schemas import Assembly
+
+from fusor.fusor import FUSOR
+from fusor.models import (
+    Assay,
+    AssayedFusion,
+    CausativeEvent,
+    EventType,
+    GeneElement,
+    TranscriptSegmentElement,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class Caller(str, Enum):
+    """Define different supported callers"""
+
+    JAFFA = "JAFFA"
+    STAR_FUSION = "STAR-Fusion"
+    FUSION_CATCHER = "FusionCatcher"
+    FUSION_MAP = "FusionMap"
+    ARRIBA = "Arriba"
+    CICERO = "CICERO"
+    MAPSPLICE = "MapSplice"
+    ENFUSION = "EnFusion"
+    GENIE = "GENIE"
+
+
+class Translator:
+    """Class for translating outputs from different fusion detection algorithms
+    to FUSOR AssayedFusion objects
+    """
+
+    def __init__(self, fusor: FUSOR) -> None:
+        """Initialize Translator class
+
+        :param fusor: A FUSOR instance
+        """
+        self.fusor = fusor
+
+    def _format_fusion(
+        self,
+        gene_5prime: GeneElement,
+        gene_3prime: GeneElement,
+        tr_5prime: TranscriptSegmentElement | None = None,
+        tr_3prime: TranscriptSegmentElement | None = None,
+        ce: CausativeEvent | None = None,
+        rf: bool | None = None,
+        assay: Assay | None = None,
+    ) -> AssayedFusion:
+        """Format classes to create AssayedFusion objects
+
+        :param gene_5prime: 5'prime GeneElement
+        :param gene_3prime: 3'prime GeneElement
+        :param Optional[tr_5prime]: 5'prime TranscriptSegmentElement
+        :param Optional[tr_3prime]: 3'prime TranscriptSegmentElement
+        :param Optional[ce]: CausativeEvent
+        :param Optional[rf]: A boolean indicating if the reading frame is preserved
+        :param Optional[assay]: Assay
+        :return AssayedFusion object
+        """
+        params = {}
+        if not tr_5prime[0] and not tr_3prime[0]:
+            params["structure"] = [gene_5prime, gene_3prime]
+        elif tr_5prime[0] and not tr_3prime[0]:
+            params["structure"] = [tr_5prime[0], gene_3prime]
+        elif not tr_5prime[0] and tr_3prime[0]:
+            params["structure"] = [gene_5prime, tr_3prime[0]]
+        else:
+            params["structure"] = [tr_5prime[0], tr_3prime[0]]
+
+        if ce:
+            params["causativeEvent"] = ce
+        if rf:
+            params["readingFramePreserved"] = rf
+        if assay:
+            params["assay"] = assay
+        return AssayedFusion(**params), None
```

Comment:
this is returning a tuple (AssayedFusion, None). I think you don't want the `None` param, just return the fusion.

We use this pattern in other places to return details about errors, but if you're always just returning None for that then there's no point in including it